src/block-components/typography/style.js

```diff
@@ -0,0 +1,204 @@
+					}
+					if ( getAttribute( 'fontSize', device, state ) !== '' ) {
+						value = `${ getAttribute( 'fontSize', device, state ) }${ getAttribute( 'fontSizeUnit', device, state ) || 'px' }`
+					}
+				}
+
+				if ( device === 'mobile' ) {
+					if ( clampDesktopValue ) {
+						value = `${ clampDesktopValue }${ getAttribute( 'fontSizeUnit', device, state ) || 'px' }`
+					}
+
+					if ( clampTabletValue ) {
+						value = `${ clampTabletValue }${ getAttribute( 'fontSizeUnit', device, state ) || 'px' }`
+					} else if ( clampDesktopValue || getAttribute( 'fontSize', device, state ) !== '' ) {
+						value = undefined
+					}
+
+					if ( getAttribute( 'fontSize', device, state ) !== '' ) {
+						value = `${ getAttribute( 'fontSize', device, state ) }${ getAttribute( 'fontSizeUnit', device, state ) || 'px' }`
+					}
+				}
+
+				return value
+			},
+			dependencies: [ 'fontSizeUnit', 'fontSize' ],
+		},
+		{
+			selector,
+			styleRule: 'color',
+			hover: 'all',
+			hoverSelector,
+			attrName: 'textColor1',
+			valuePreCallback: ( value, getAttribute, device, state ) => {
+				if ( getAttribute( 'textColorType', 'desktop', state ) === 'gradient' ) {
+					return undefined
+				}
+				return value
+			},
+			dependencies: [ 'textColorType' ],
+		},
+		{
+			selector,
+			styleRule: 'backgroundImage',
+			hover: 'all',
+			hoverSelector,
+			attrName: 'textColor1',
+			valuePreCallback: ( value, getAttribute, device, state ) => {
+				if (
+					getAttribute( 'textColorType', 'desktop', state ) !== 'gradient' ||
+					getAttribute( 'textColor1', 'desktop', state ) === '' ||
+					getAttribute( 'textColor2', 'desktop', state ) === ''
+				) {
+					return undefined
+				}
+				return value
+			},
+			valueCallback: ( value, getAttribute, device, state ) => {
+				const textColor1 = getAttribute( 'textColor1', 'desktop', state )
+				const textColor2 = getAttribute( 'textColor2', 'desktop', state )
+				const textGradientDirection = getAttribute( 'textGradientDirection', 'desktop', state )
+
+				return `linear-gradient(${ textGradientDirection !== '' ? `${ textGradientDirection }deg, ` : '' }${ textColor1 }, ${ textColor2 })`
+			},
+			dependencies: [ 'textColorType', 'textColor1', 'textColor2', 'textGradientDirection' ],
+		},
+		{
+			selector,
+			styleRule: 'textAlign',
+			attrName: 'textAlign',
+			hover: 'all',
+			hoverSelector,
+			responsive: 'all',
+		},
+		{
+			selector,
+			styleRule: 'lineHeight',
+			attrName: 'lineHeight',
+			hover: 'all',
+			hoverSelector,
+			responsive: 'all',
+			hasUnits: 'em',
+		},
+		{
+			selector,
+			styles: {
+				fontWeight: 'fontWeight',
+				textTransform: 'textTransform',
+			},
+			hover: 'all',
+			hoverSelector,
+		},
+		{
+			selector,
+			styleRule: 'fontFamily',
+			attrName: 'fontFamily',
+			valueCallback: value => getFontFamily( value ),
+		},
+		{
+			selector,
+			styleRule: 'letterSpacing',
+			attrName: 'letterSpacing',
+			hasUnits: 'px',
```

Comment:
`hasUnits` is used alongside `hover="all"` and `stkHover`. Since you're using this to format the CSS rule, use `format: '%spx'` instead.